packages/odyssey-react-mui/src/FieldSet.tsx

```diff
@@ -0,0 +1,66 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { memo, ReactElement } from "react";
+import { Infobox, Typography } from "./";
+
+export interface FieldSetProps {
+  /**
+   * The title of the FieldSet
+   */
+  legend: string;
+  /**
+   * A supplementary description
+   */
+  description?: string;
+  /**
+   * The Field components within the FieldSet
+   */
+  children: ReactElement | Array<ReactElement>;
+  /**
+   * An Infobox indicating a FieldSet-wide error or status update.
+   */
+  alert?: ReactElement<typeof Infobox>;
+  /**
+   * The name associated with the group.
+   */
+  name: string;
+  /**
+   * If this Boolean attribute is set, all form controls that are descendants of the <fieldset>, are disabled, meaning they are not editable and won't be submitted along with the <form>.
+   * They won't receive any browsing events, like mouse clicks or focus-related events.
+   * Note that form elements inside the <legend> element won't be disabled.
+   */
+  isDisabled?: boolean;
+}
+
+const FieldSet = ({
```

Comment:
As we spoke in chat:
```suggestion
const Fieldset = ({
```